packages/components/text/src/text.tsx

```diff
@@ -0,0 +1,248 @@
+import type { MessageDescriptor } from 'react-intl';
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import requiredIf from 'react-required-if';
+import { useTheme } from '@emotion/react';
+import isNil from 'lodash/isNil';
+import { oneLine } from 'common-tags';
+import {
+  filterDataAttributes,
+  warnDeprecatedProp,
+} from '@commercetools-uikit/utils';
+import {
+  bodyStyles,
+  detailStyles,
+  headlineStyles,
+  subheadlineStyles,
+  wrapStyles,
+} from './text.styles';
+
+type TTextProps = {
+  intlMessage?: MessageDescriptor;
+  children?: React.ReactNode;
+};
+
+type TDeprecatedProps = {
+  // @deprecated
+  elementType: string;
+};
+
+const warnIfNoTextProps = (props: TTextProps, componentName) => {
+  if (!props.intlMessage && !React.Children.count(props.children)) {
+    console.error(
+      oneLine`
+        Warning: Failed prop type:
+        The prop \`intlMessage\` is marked as required in \`${componentName}\`,
+        but its value is \`undefined\`
+      `
+    );
+    console.error(
+      oneLine`
+        Warning: Failed prop type:
+        The prop \`children\` is marked as required in \`${componentName}\`,
+        but its value is \`undefined\`
+      `
+    );
+  }
+};
+
+const Text = (props: TTextProps) => (
+  <>
+    {props.intlMessage ? (
+      <FormattedMessage {...props.intlMessage} />
+    ) : (
+      props.children
+    )}
+  </>
+);
+Text.displayName = 'Text';
+
+type THeadlineProps = {
+  as?: 'h1' | 'h2' | 'h3';
+  // @deprecated
+  elementType?: 'h1' | 'h2' | 'h3';
+  title?: string | null;
```

Comment:
Maybe extract this into `TNonEmptyString`? 🤔

Reply:
what do want to accomplish with `TNonEmptyString`?
This could just be `title: string` actually, because otherwise the approach is to emit an error to the log.

Or am I missing something?